### ISSUE TITLE
[SQLite] `dropForeign` is silently ignored

### ISSUE THREAD

**Issue by rijkvanzanten** [knex/knex#4369]
# Environment

Knex version: 0.95.2
Database + version: SQLite (any)
OS: macOS

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

When using [`.dropForeign()`](http://knexjs.org/#Schema-dropForeign) in SQLite, it's silently ignored. When inspecting the queries made, it correctly creates a new temp table for the `alterTable` call, but fails to strip out the `FOREIGN KEY` line in the create table sql.

2. Error message

n/a

3. Reduced test code

```js
require('sqlite3');
const Knex = require('knex');

const sqlite = Knex({
  client: 'sqlite',
  connection: ':memory:',
});

(async function run() {
  await sqlite.schema.createTable('test', (t) => {
    t.increments('id').primary();
  });
  
  await sqlite.schema.createTable('related', (t) => {
    t.increments('id').primary();
    t.integer('fk').references('id').inTable('test');
  });
  
  console.log(await getCreateTableSQL('related'));
  // Contains the `FOREIGN KEY` as expected
  
  await sqlite.schema.alterTable('related', (t) => {
    t.dropForeign(['fk']);
  });
  
  console.log(await getCreateTableSQL('related'));
  // Still contains the `FOREIGN KEY`. Should've been removed.
})();

async function getCreateTableSQL(table) {
    const result = await sqlite
        .select('sql')
        .from('sqlite_master')
        .where({ name: table })
        .first();
    
    return result && result.sql;
}
```

https://runkit.com/rijkvanzanten/knex-sqlite-drop-foreign-example

(@nickrum, before I start chopping away in your amazing work from the other day, do you have any instructions on how to (re-)implement `dropForeign` using the new tokenized approach you set up?)

**Comment by rijkvanzanten** [comment]
I'm also a little confused why the tests haven't caught this 🤔 

There is [an integration test for foreign keys](https://github.com/knex/knex/blob/master/test/integration/schema/foreign-keys.js) that does seem to test for this:

https://github.com/knex/knex/blob/2bce36e3eef467354f47370802379fcac56d1084/test/integration/schema/foreign-keys.js#L80-L91

but it still passes in SQLite..

I did see that foreign_keys is enabled in the pool options in the test-runner:

https://github.com/knex/knex/blob/2bce36e3eef467354f47370802379fcac56d1084/test/knexfile.js#L24-L32

The only thing I can think of is that this option somehow isn't actually used? That would explain SQLite silently ignoring foreign key constraints (@kibertoad)

**Comment by nickrum** [comment]
I think the issue with the integration test is that it doesn't actually check if the foreign key constraints have been deleted.

**Comment by rijkvanzanten** [comment]
I'd expected that insertion on 86 to throw an invalid key constraint error though, seeing it didn't delete the foreign key constraint 🤔

**Comment by nickrum** [comment]
Yeah right, but the test would pass anyway without an `expect` assertion, wouldn't it? 🤔

**Comment by rijkvanzanten** [comment]
Hmm I was under the impression that an unexpected error (in this case the throw from `insert`) would mark the test as failed. You might be right tho!

**Comment by nickrum** [comment]
I just checked it and it seems like tests indeed pass even if there was an exception. Mocha even says `No unhandled exceptions`. Maybe this is due to the test being asynchronous.

But in this case the `insert` doesn't throw an exception because apparently if you drop a composite foreign key by specifying the individual columns, all foreign keys are dropped.

**Comment by rijkvanzanten** [comment]
@kibertoad I hate to be _that guy_, but when is the next patch release scheduled? 😁 
This fix is a pretty massive deal for us, as our SQLite migrations are totally bust 👀

**Comment by kibertoad** [comment]
@rijkvanzanten Apologies for the delay, another swamp surge came in and consumed me entirely. Appreciate the ping, will do the release tonight.

**Comment by rijkvanzanten** [comment]
No worries! I totally understand the business and _life_ getting in the way 😄

**Comment by kibertoad** [comment]
Released in 0.95.3.

**Comment by tungrix** [comment]
The `dropForeign()` is not always silently ignored.

But if the related column was added with inline reference, the subsequent dropForeign will be silently ignored.

For example, in migration 1:
```typescript
export async function up(knex: Knex): Promise<void> {
  await knex.raw('alter table `event` add column `sports_id` integer not null references `sports`(`id`)')
}
```

in migration 2:
```typescript
export async function up(knex: Knex): Promise<void> {
  await knex.schema.alterTable('event', table => {
    table.dropForeign('sports_id')
  })
}
```

In this example, the foreign is still there. We even tried to drop the sports table without error.

But subsequent insert will fail since it will check for the foreign key constraint, and the check will fail because the referred sports table no longer exist.
